crates/ef-testing/src/evm_sequencer/evm_state.rs

```diff
@@ -142,12 +143,48 @@ impl EvmState for KakarotSequencer {
         storage.append(allowance_storage);
 
         // Write all the storage vars to the sequencer state.
-        let starknet_address = starknet_address.try_into()?;
         for (k, v) in storage {
-            (&mut self.0.state).set_storage_at(starknet_address, k, v);
+            (&mut self.0.state).set_storage_at(*FEE_TOKEN_ADDRESS, k, v);
         }
         Ok(())
     }
+
+    fn get_storage_at(&mut self, evm_address: &Address, key: U256) -> StateResult<U256> {
+        let keys = split_u256(key).map(Into::into);
+        let keys = get_uint256_storage_var_addresses("storage_", &keys).unwrap(); // safe unwrap: all vars are ASCII
```

Comment:
this cannot work on EOAs right? 
what will happen in case the evm_address from the params is an EOA, or a non-deployed EVM address?